save-backend/src/main/kotlin/org/cqfn/save/backend/service/ExecutionService.kt

```diff
@@ -77,4 +80,18 @@ class ExecutionService(private val executionRepository: ExecutionRepository) {
      */
     fun getLatestExecutionByProjectNameAndProjectOwner(name: String, owner: String): Execution? =
             executionRepository.findTopByProjectNameAndProjectOwnerOrderByStartTimeDesc(name, owner)
+
+    /**
+     * @param executionInitializationDto execution dto to update
+     * @return execution
+     */
+    fun updateNewExecution(executionInitializationDto: ExecutionInitializationDto) =
+            executionRepository.findTopByProjectOrderByStartTimeDesc(executionInitializationDto.project)?.let {
+                assert(it.version == null) { "Execution was already updated" }
```

Comment:
Use `require` on something similar for preconditions; it's better suited for this than JVM-specific `assert`. Also, you still need `-ea` flag even for newer JVMs to enable asserts